optimizely/event/processor.go

```diff
@@ -110,6 +110,10 @@ func (p *QueueingEventProcessor) StartTicker(ctx context.Context) {
 			case <-ctx.Done():
 				pLogger.Debug("Event processor stopped, flushing events.")
 				p.FlushEvents()
+				d, ok := p.EventDispatcher.(*QueueEventDispatcher)
+				if ok {
+					d.flushEvents()
```

Comment:
I think `FlushEvents` should be part of the dispatcher interface. If the user provides their own, wouldn't we want to be able to flush their events on cancel?